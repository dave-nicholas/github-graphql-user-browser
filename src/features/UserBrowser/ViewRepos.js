import React from 'react';
import PropTypes from 'prop-types';

const ViewRepos = ({ repositories: { edges } }) =>
  !!edges.length ? (
    <ul className="collection">
      {edges.map(({ node: { url, name, description } }, i) => (
        <li className="collection-item" key={i}>
          <span className="title">
            <a href={url}>{name}</a>
          </span>
          <p>{description}</p>
        </li>
      ))}
    </ul>
  ) : (
    'No repos...'
  );

ViewRepos.propTypes = {
  repositories: PropTypes.shape({
    edges: PropTypes.arrayOf(
      PropTypes.shape({
        url: PropTypes.string,
        name: PropTypes.string,
        description: PropTypes.string
      })
    )
  }).isRequired
};

export default ViewRepos;
