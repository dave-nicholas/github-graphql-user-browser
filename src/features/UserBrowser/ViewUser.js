import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

const ViewUser = ({ user, avatarUrl, createdAt }) => {
  return (
    <div className="card">
      <div className="card-image">
        <img src={avatarUrl} alt={user} />
      </div>
      <div className="card-stacked">
        <div className="card-content">
          User since: {moment(createdAt).format('MMMM Do YYYY')}
        </div>
      </div>
    </div>
  );
};

ViewUser.propTypes = {
  avatarUrl: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
  user: PropTypes.string.isRequired
};

export default ViewUser;
