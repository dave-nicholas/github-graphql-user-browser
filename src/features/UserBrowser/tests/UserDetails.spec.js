import React from 'react';
import { shallow } from 'enzyme';
import UserDetailConnected, {
  UserDetails,
  UserDetailQuery
} from '../UserDetails';
import ViewRepos from '../ViewRepos';
import ViewUser from '../ViewUser';
import Loader from '../../../components/Loader';
import toJson from 'enzyme-to-json';

const mockData = {
  user: {
    avatarUrl: '//test.com/',
    createdAt: '2011-05-25T18:18:31Z',
    repositories: {}
  }
};

describe('UserDetails', () => {
  const wrapper = shallow(<UserDetails data={mockData} username="test" />);

  it('should render loader', () => {
    expect(
      shallow(
        <UserDetails
          data={{
            loading: true
          }}
          username="test"
        />
      ).contains(<Loader />)
    ).toBe(true);
  });

  it('should render ViewRepos', () => {
    expect(wrapper.contains(<ViewRepos repositories={{}} />)).toBe(true);
  });

  it('should render ViewUsers', () => {
    expect(
      wrapper.contains(
        <ViewUser
          avatarUrl="//test.com/"
          createdAt="2011-05-25T18:18:31Z"
          user="test"
        />
      )
    ).toBe(true);
  });

  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('UserDetailQuery', () => {
  it('should match snapshot', () => {
    expect(UserDetailQuery).toMatchSnapshot();
  });
});
