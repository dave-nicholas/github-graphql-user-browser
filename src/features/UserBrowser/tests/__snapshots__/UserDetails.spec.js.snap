// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetailQuery should match snapshot 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": undefined,
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "login",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "username",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "user",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "avatarUrl",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "createdAt",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "last",
                      },
                      "value": Object {
                        "kind": "IntValue",
                        "value": "5",
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "repositories",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "edges",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "node",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "url",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "name",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "description",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "username",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 252,
    "start": 0,
  },
}
`;

exports[`UserDetails should match snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col s12 m6"
  >
    <ViewUser
      avatarUrl="//test.com/"
      createdAt="2011-05-25T18:18:31Z"
      user="test"
    />
  </div>
  <div
    className="col s12 m6"
  >
    <ViewRepos
      repositories={Object {}}
    />
  </div>
</div>
`;
